src/librustc/traits/object_safety.rs

```diff
@@ -47,13 +47,15 @@ impl ObjectSafetyViolation {
                 "the trait cannot use `Self` as a type parameter \
                  in the supertraits or where-clauses".into(),
             ObjectSafetyViolation::Method(name, MethodViolationCode::StaticMethod) =>
-                format!("method `{}` has no receiver", name).into(),
-            ObjectSafetyViolation::Method(name, MethodViolationCode::ReferencesSelf) =>
-                format!("method `{}` references the `Self` type \
-                         in its arguments or return type", name).into(),
-            ObjectSafetyViolation::Method(name,
-                                            MethodViolationCode::WhereClauseReferencesSelf(_)) =>
-                format!("method `{}` references the `Self` type in where clauses", name).into(),
+                format!("associated function `{}` has no `self` parameter", name).into(),
+            ObjectSafetyViolation::Method(name, MethodViolationCode::ReferencesSelf) => format!(
+                "method `{}` references the `Self` type in its arguments or return type",
+                name,
+            ).into(),
+            ObjectSafetyViolation::Method(
+                name,
+                MethodViolationCode::WhereClauseReferencesSelf(_),
+            ) => format!("method `{}` references the `Self` type in where clauses", name).into(),
```

Comment:
This one doesn't seem to either.

Reply:
src/test/ui/issues/issue-50781.stderr should have been updated, running `--bless` again